src/Findologic/Config/FindologicConfigService.php

```diff
@@ -0,0 +1,307 @@
+        }
+        if ($salesChannelId && !Uuid::isValid($salesChannelId)) {
+            throw new InvalidUuidException($salesChannelId);
+        }
+        if ($languageId && !Uuid::isValid($languageId)) {
+            throw new InvalidUuidException($languageId);
+        }
+    }
+
+    private function getId(string $key, string $salesChannelId, string $languageId): ?string
+    {
+        $criteria = $this->buildCriteriaWithKey($key, $salesChannelId, $languageId);
+        $ids = $this->finSearchConfigRepository->searchIds($criteria, Context::createDefaultContext())->getIds();
+
+        return array_shift($ids);
+    }
+
+    private function buildCriteriaWithKey(string $key, string $salesChannelId, string $languageId): Criteria
+    {
+        $criteria = new Criteria();
+        $criteria->addFilter(
+            new EqualsFilter('configurationKey', $key),
+            new EqualsFilter('salesChannelId', $salesChannelId),
+            new EqualsFilter('languageId', $languageId)
+        );
+
+        return $criteria;
+    }
+
+    private function load(string $salesChannelId, string $languageId): array
+    {
+        $key = sprintf('%s-%s', $salesChannelId, $languageId);
+
+        if (isset($this->configs[$key])) {
+            return $this->configs[$key];
+        }
+
+        $criteria = new Criteria();
+        if (method_exists($criteria, 'setTitle')) {
+            $criteria->setTitle('finsearch-config::load');
+        }
+
+        $criteria->addFilter(
+            new MultiFilter(
+                MultiFilter::CONNECTION_AND,
+                [
+                    new EqualsFilter('salesChannelId', $salesChannelId),
+                    new EqualsFilter('languageId', $languageId),
+                ]
+            )
+        );
+
+        $criteria->addSorting(
+            new FieldSorting('salesChannelId', FieldSorting::ASCENDING),
+            new FieldSorting('id', FieldSorting::ASCENDING)
+        );
+        $criteria->setLimit(500);
+
+        $configs = new FinSearchConfigCollection();
+        $iterator = new RepositoryIterator(
+            $this->finSearchConfigRepository,
+            Context::createDefaultContext(),
+            $criteria
+        );
+
+        while ($chunk = $iterator->fetch()) {
+            $configs->merge($chunk->getEntities());
+        }
+
+        $this->configs[$key] = $this->buildFindologicConfigArray($configs);
+
+        return $this->configs[$key];
+    }
+
+    private function buildFindologicConfigArray(FinSearchConfigCollection $configs): array
+    {
+        $configValues = [];
+
+        foreach ($configs as $config) {
+            $keys = explode('.', $config->getConfigurationKey());
+
+            $configValues = $this->getSubArray($configValues, $keys, $config->getConfigurationValue());
+        }
+
+        return $configValues;
+    }
+
+    private function getSubArray(array $configValues, array $keys, $value): array
```

Comment:
Adding type information inside of a method doesn't make much sense.
```suggestion
    private function getSub(array $configValues, array $keys, $value): array
```

This method is really weird, as it requires in-depth knowledge on how to use it. I wouldn't know what "getSubArray" does, even if I would check all method arguments. Rename this method and add some kind of docblock that makes understanding this method easier.